code/client/munkilib/updatecheck.py

```diff
@@ -1030,6 +1030,25 @@ def compare_version_keys(a, b):
                         rejected_items.append(reason)
                         continue
 
+                if item.get('deploy_percent', ''):
+                    deploy_percent = item['deploy_percent']
+                    munkicommon.display_debug1(
+                        'Considering item %s, version %s '
+                        'with %s deploy percent required',
+                        item['name'], item['version'], deploy_percent)
+                    munkicommon.display_debug1(
+                        'Machine deploy percent is %s', MACHINE['shard'])
+                    if (MACHINE['shard'] > deploy_percent):
+                        # skip this one, go to the next
+                        munkicommon.display_debug1(
+                            'Item %s, version %s is set to deploy to machines '
+                            'with a shard value of %s or lower. '
+                            'Our shard value is %s.'
+                            % (item['name'], item['version'],
```

Comment:
It's not a range though - it's zero through whatever is specified.

In this or any similar form it should be called deploy percentage, as it's
not shards. Shards would be akin to A/B testing through defined groups which
this does not do.

On Tuesday, November 24, 2015, Mike Dodge <notifications@github.com> wrote:

> In code/client/munkilib/updatecheck.py
> <https://github.com/munki/munki/pull/547#discussion_r45786503>:
>
> > @@ -1030,6 +1030,25 @@ def compare_version_keys(a, b):
> >                          rejected_items.append(reason)
> >                          continue
> >
> > +                if item.get('deploy_percent', ''):
> > +                    deploy_percent = item['deploy_percent']
> > +                    munkicommon.display_debug1(
> > +                        'Considering item %s, version %s '
> > +                        'with %s deploy percent required',
> > +                        item['name'], item['version'], deploy_percent)
> > +                    munkicommon.display_debug1(
> > +                        'Machine deploy percent is %s', MACHINE['shard'])
> > +                    if (MACHINE['shard'] > deploy_percent):
> > +                        # skip this one, go to the next
> > +                        munkicommon.display_debug1(
> > +                            'Item %s, version %s is set to deploy to machines '
> > +                            'with a shard value of %s or lower. '
> > +                            'Our shard value is %s.'
> > +                            % (item['name'], item['version'],
>
> What about deploy_percent_range, deploy_percent ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/munki/munki/pull/547/files#r45786503>.
>


-- 
Samuel Keeley